src/OpenCensus.Collector.Dependencies/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -0,0 +1,119 @@
+        {
+            if (!(this.startRequestFetcher.Fetch(payload) is HttpRequestMessage request))
+            {
+                // Debug.WriteLine("request is null");
+                return;
+            }
+
+            // TODO: this needs to be generilized
+            if (request.RequestUri.ToString().Contains("zipkin.azurewebsites.net"))
+            {
+                return;
+            }
+
+            this.Tracer.SpanBuilder("HttpOut").SetSampler(this.Sampler).StartScopedSpan();
+            var span = this.Tracer.CurrentSpan;
+            span.PutClientSpanKindAttribute();
+            span.PutHttpMethodAttribute(request.Method.ToString());
+            span.PutHttpHostAttribute(request.RequestUri.Host);
+            span.PutHttpPathAttribute(request.RequestUri.AbsolutePath);
+            span.PutHttpUrlAttribute(request.RequestUri.ToString());
+        }
+
+        public override void OnStopActivity(Activity activity, object payload)
+        {
+            var span = this.Tracer.CurrentSpan;
+
+            if (!(this.stopResponseFetcher.Fetch(payload) is HttpResponseMessage response))
+            {
+                // response could be null for DNS issues, timeouts, etc...
+                // TODO: how do we make sure we will not close a scope that wasn't opened?
+
+                span.End();
+                return;
+            }
+
+            var requestTaskStatus = this.stopRequestStatusFetcher.Fetch(payload) as TaskStatus?;
+
+            if (requestTaskStatus.HasValue)
+            {
+                if (requestTaskStatus != TaskStatus.RanToCompletion)
+                {
+                    span.PutErrorAttribute();
+
+                    if (requestTaskStatus == TaskStatus.Canceled)
+                    {
+                        span.Status = Status.Cancelled;
+                    }
+                }
+            }
+
+            span.PutHttpStatusCode((int)response.StatusCode, response.ReasonPhrase);
+
+            span.End();
+        }
+
+        public override void OnException(Activity activity, object payload)
+        {
+            if (!(this.stopExceptionFetcher.Fetch(payload) is Exception exc))
+            {
+                // Debug.WriteLine("response is null");
+                return;
+            }
+
+            var span = this.Tracer.CurrentSpan;
+
+            if (exc is HttpRequestException)
+            {
+                // TODO: on netstandard this will be System.Net.Http.WinHttpException: The server name or address could not be resolved
+                if (exc.InnerException is WebException &&
+                    ((WebException)exc.InnerException).Status == WebExceptionStatus.NameResolutionFailure)
+                {
+                    span.Status = Status.InvalidArgument;
+                }
+                else if (exc.InnerException != null)
+                {
+                    span.Status = Status.Unknown.WithDescription(exc.Message);
```

Comment:
this probably conflicts with status assignment in OnStop. We should do it just here or just there

Reply:
it's OK to override it from the response code. This code runs for DNS error for instance

Reply:
oh, i see you override just the status, but not error message